demos/blockfactory/app_controller.js

```diff
@@ -137,9 +140,12 @@ AppController.prototype.exportBlockLibraryToFile = function() {
   // Download file if all necessary parameters are provided.
   if (filename) {
     FactoryUtils.createAndDownloadFile(blockLibText, filename, 'xml');
+    BlocklyDevTools.Analytics.onExport('BlockFactoryLibrary.xml');
```

Comment:
The strings that indicate the type should be constants defined in BlocklyDevTools.Analytics.  That can be done in a followup as you flesh this out.